src/app/(dashboard)/(routes)/languages/components/language-dashboard.tsx

```diff
@@ -29,36 +22,83 @@ type Props = {
     points: number;
   }[];
   lang: string;
+  langId: number;
 };
 
-const LanguageDashboard = ({ levels, lang }: Props) => {
-  const [leaderboardData, setLeaderboardData] = useState({
-    studentMeta: {},
-    topTenStudentList: [],
+const LanguageDashboard = ({ levels, lang, langId }: Props) => {
+  const userId = getCookie("userId");
+
+  const { data: dashboardData, isLoading: dashboardLoader } =
+    useQuery<LeaderboardResponse>({
+      queryKey: ["languageLeaderboard", userId, langId],
+      queryFn: () =>
+        getLanguageDashboard({
+          userId: userId || null,
+          lang: langId || null,
+        }),
+      enabled: !!userId && !!langId,
+    });
+
+  const { data: activityData, isLoading: activityLoader } = useQuery({
+    queryKey: ["studentActivity", userId, langId],
+    queryFn: () =>
+      getStudentActivity({
+        studentId: userId || null,
+        subjectId: langId || null,
+      }),
+    enabled: !!userId && !!langId,
   });
-  const [studentActivity, setStudentActivity] = useState([]);
-  const [streakData, setStreakData] = useState({});
-  const [studentData, setStudentData] = useState(null);
-  const [topicData, setTopicData] = useState<TopicCardLayoutArr>([]);
+
   const [avatar, setAvatar] = useState<string>("");
-  const [topicLoader, setTopicLoader] = useState<boolean>(false);
-  const [dashboardLoader, setDashboardLoader] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (activityData?.response) {
+      if (activityData.response.currentStudentMeta?.pic) {
+        setAvatar(activityData.response.currentStudentMeta.pic);
+        localStorage.setItem(
+          "user-avatar",
```

Comment:
This is set but not used anywhere, why is this required

Reply:
avatar is being used on quiz pages, and I didn't add it